src/index.js

```diff
@@ -194,3 +265,82 @@ function exec_git(args, options) {
       });
   }));
 }
+
+function addOptionsFromCommandLineArguments(options) {
+  const parser = new argparse.ArgumentParser({
+    version,
+    addHelp: true,
+    description: 'Clones git repositories from a Leeroy configuration.'
+  });
+
+  const subparsers = parser.addSubparsers({
+    title: 'subcommands',
+    dest: 'command'
+  });
+
+  const commands = createCommands();
+  commands.map(x => { x.initializeParsers(parser, subparsers); });
+
+  const args = parser.parseArgs();
+  commands.map(x => {
+    if (x.name !== args.command) return;
+    if (x.validateArgs) x.validateArgs(args);
+    x.addOptions(args, options);
+  });
+}
+
+function createCommands()
+{
+  return [
+    {
+      name: 'clone',
+      initializeParsers: (parser, subparsers) => {
+        const cloneParser = subparsers.addParser('clone', {
+          addHelp: true,
+          help: `Clones a configuration. The configuration name will be saved to the local ${configFileName} file.  If the file exists, it will be overwritten.`
+        });
+        cloneParser.addArgument(['config'], {
+          help: 'The Leeroy configuration name.'
+        });
+      },
+      addOptions: (args, options) => {
+        options.configName = args.config;
+        options.save = true;
+      }
+    },
+    {
+      name: 'rollback',
```

Comment:
That's a good point.  I thought about using `checkout` but it seemed too similar to `clone,` and a bit ambiguous in that context. However, given the two subcommands/options I ended up with, subcommands might be a little overkill.

`clone-leeroy [configName] [-t | -d | -r] [--save]` would be a bit simpler and still provide the same functionality (with the added benefit of not forcing the user to save a .clonejs file).